### ISSUE TITLE
Ubuntu 18.04 LTS Support

### ISSUE THREAD

**Issue by ryanRfox** [bitshares/bitshares-core#835]
## User Story
As an `Ubuntu User` I want to `build bitshares-core on the (forthcoming) Ubuntu 18.04 LTS platform` because it is the `latest version`.

## Notes
- Ubuntu 18.04 LTS (bionic)  [Release Notes](https://wiki.ubuntu.com/BionicBeaver/ReleaseNotes)
- Linux kernel [4.15](https://wiki.ubuntu.com/BionicBeaver/ReleaseNotes#Linux_kernel_4.15) 
- openssl 1.1.0g ([apt install openssl](https://packages.ubuntu.com/bionic/openssl))
- boost 1.65.1 ([apt install libboost-all-dev](https://packages.ubuntu.com/bionic/libboost-all-dev))

## Task List
- [x] Upgrade to 18.04 (until final release 28 APR 2018)
  - Prerequisite: `apt install update-manager-core`
  - Verify: `/etc/update-manager/release-upgrades` is set to `normal`
  - Install: `do-release-upgrade -d`
  - follow on screen instructions
- [x] Add support for OpenSSL 1.1 (see branch [openssl-1.1-support](https://github.com/bitshares/bitshares-core/tree/openssl-1.1-support))
- [x] Add support for Boost 1.65
- [x] Update build script(s)
  - See @jmjatlanta comment below for complete build script
- [x] Test building on Ubuntu 18.04 LTS
  - [x] README.md: PR #1212 
  - [x]  Wiki: Update on day of release
    - [x] wiki home page
    - [x] wiki BUILD_UBUNTU page
    - [x] bump docs submodule after wiki updated.
- [x] Update documentation

**Comment by jmjatlanta** [comment]
I was able to compile and run bitshares-core on Ubuntu 18.04 (latest daily build) with the following caveats:
1. The bitshares-fc and bitshares-core branch openssl-1.1-support was used
2. The changes in bitshares-fc branch boost_1.66 was also merged in.

With a fresh install of Ubuntu 18.04, the following additional packages are necessary:
``sudo apt install gcc g++ git cmake libboost-all-dev libssl-dev autoconf libcurl4-openssl-dev``

Once the openssl 1.1 and boost 1.66 changes are merged in to bitshares-fc and bitshares-core, it will be as simple as:

``git clone https://github.com/bitshares/bitshares-core.git``
``cd bitshares-core``
``cmake -DCMAKE_BUILD_TYPE=Release .``
``make``

**Comment by abitmore** [comment]
While this is a good to have feature and the correct direction to go, I don't think it's very high priority. After Ubuntu 18.04 is out, it may need some time for businesses to try it in production. My estimation on this is about 3-6 months.

**Comment by operano02** [comment]
hi ;
can you explane the  " boost_1.65 branches are merged in to bitshares-fc and bitshares-core, "
what is merged into ? how to merge into ? i'm newbie linux user
after "  git pull https://github.com/jmjatlanta/bitshares-fc.git boost_1.65 " command
" fatal: Couldn't find remote ref boost_1.65 " this error.

**Comment by jmjatlanta** [comment]
The work done on the boost_1.65 has now become part of the official bitshares-fc project. So, that line can be left out. I will adjust the post above.

**Comment by jmjatlanta** [comment]
To compile for the latest release of TESTNET ( instructions as of 12-May-2018 )

``# Pull down the bitshares-core repository and place it in the directory ``
``# bitshares-core (directory should not already exist)``
``git clone https://git@github.com/bitshares/bitshares-core``
``# Enter your new copy of the repository``
``cd bitshares-core``
``# checkout the testnet branch of bitshares-core``
``git checkout testnet``
``# checkout all the submodules needed``
``git submodule update --init --recursive``
``# merge in the openssl-1.1-support branch of bitshares-core``
``git pull https://github.com/bitshares/bitshares-core openssl-1.1-support``
``# move to the bitshares-fc directory ``
``cd libraries/fc``
``# grab the latest, if you want to ( this is optional )``
``git checkout master``
``# merge in the openssl-1.1-support branch of bitshares-fc``
``git pull https://github.com/bitshares/bitshares-fc.git openssl-1.1-support``
``# go back to the bitshares-core directory ``
``cd ../..``
``# prepare your Makefiles ( as you normally would )``
``cmake -DCMAKE_BUILD_TYPE=Release .``
``# compile the applications ( as you normally would )``
``make``

**Comment by abitmore** [comment]
@jmjatlanta that means the only thing we need to do is to merge the OpenSSL 1.1 PR's?

We need to test:
* sync testnet and mainnet,
* connect a light wallet and make some transactions,
* sign blocks and
* etc.

**Comment by jmjatlanta** [comment]
@abitmore I have not tested the OpenSSL-1.1-support branch completely. 

I would add that we need to test compiling with OpenSSL 1.0.2 with the 1.1-support branch merged in.

If we can get everything tested well and the 1.1-support branch merged in to the other branches, it would be a good win for those running 18.04 LTS, as they would not need anything beyond stuff they can apt-get plus the bitshares-core branch of their choice.

**Comment by ryanRfox** [comment]
I am failing to build at the same point with both OpenSSL 1.0.2g & 1.1.0g as demonstrated here:

![image](https://user-images.githubusercontent.com/13419165/39966720-b27ec13a-567e-11e8-81f5-f5a90309ca89.png)

@jmjatlanta I really appreciate the detailed instructions with inline comments above.

**Comment by pmconrad** [comment]
See #921, it includes the removal of these API calls

**Comment by ryanRfox** [comment]
Success! Yes, after commenting out the lines related to blind_sign API calls and deterministic_openssl_rand found in #921 the build completed without errors.

**Comment by FrankBlabu** [comment]
Are the instructions in the comments above still valid ? I get the reply "Couldn't find remote ref openssl-1.1-support" when trying to pull it.

I would appreciate any hint here. I updated my system to Ubuntu 18.04 without checking for the witness node compatibility before and now I'm stuck and cannot go back easily.

**Comment by pmconrad** [comment]
The branch has been merged into develop, please check out develop branch and build that. Don't forget to ``git submodule update --init --recursive``.

**Comment by ryanRfox** [comment]
@FrankBlabu you may observe the build process for the `develop` branch provided by Travis-CI:
https://travis-ci.org/bitshares/bitshares-core/builds/391476199

Focus on lines 418 & 428:
```
git clone --depth=1 --branch=develop https://github.com/bitshares/bitshares-core.git
git submodule update --init --recursive
```
As needed, please open a `Build Error` Issue and complete the template with the relevant details:
https://github.com/bitshares/bitshares-core/issues/new?template=build_error.md

**Comment by jmjatlanta** [comment]
Updated instructions as of 15-June-2018 for develop branch, based on a fresh install of Ubuntu 18.04:

``sudo apt install gcc g++ git cmake libboost-all-dev libssl-dev autoconf libcurl4-openssl-dev``
``git clone --branch=develop https://github.com/bitshares/bitshares-core.git``
``cd bitshares-core``
``git submodule update --init --recursive``
``cmake -DCMAKE_BUILD_TYPE=Release .``
``make``

**Comment by ying2025** [comment]
I have the similiar problem. But My openssl is 1.0.2. How I solve the following problems?
/home/zhr/bitshare/bitshares-core/libraries/deterministic_openssl_rand/deterministic_openssl_rand.cpp:92:1: error: invalid conversion from ‘void (*)(const void*, int)’ to ‘int (*)(const void*, int)’ [-fpermissive]
 };
 ^
/home/zhr/bitshare/bitshares-core/libraries/deterministic_openssl_rand/deterministic_openssl_rand.cpp:92:1: error: invalid conversion from ‘void (*)(const void*, int, double)’ to ‘int (*)(const void*, int, double)’ [-fpermissive]
libraries/deterministic_openssl_rand/CMakeFiles/deterministic_openssl_rand.dir/build.make:62: recipe for target 'libraries/deterministic_openssl_rand/CMakeFiles/deterministic_openssl_rand.dir/deterministic_openssl_rand.cpp.o' failed
make[2]: *** [libraries/deterministic_openssl_rand/CMakeFiles/deterministic_openssl_rand.dir/deterministic_openssl_rand.cpp.o] Error 1
CMakeFiles/Makefile2:790: recipe for target 'libraries/deterministic_openssl_rand/CMakeFiles/deterministic_openssl_rand.dir/all' failed
make[1]: *** [libraries/deterministic_openssl_rand/CMakeFiles/deterministic_openssl_rand.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
zhr@ubuntu:~/bitshare/bitshares-core$ openssl version
OpenSSL 1.0.2g  1 Mar 2016

**Comment by jmjatlanta** [comment]
Hrmmm. It seems you are having a different problem @ying2025. Please create a new issue here:

https://github.com/bitshares/bitshares-core/issues/new?template=build_error.md

**Comment by ying2025** [comment]
ubuntu:~/bitshare/bitshares-core$ git checkout openssl-1.1-support
error: pathspec 'openssl-1.1-support' did not match any file(s) known to git.

**Comment by ying2025** [comment]
[ 12%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/io/fstream.cpp.o
[ 12%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/io/sstream.cpp.o
[ 13%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/io/json.cpp.o
[ 13%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/io/varint.cpp.o
[ 13%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/io/console.cpp.o
[ 14%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/filesystem.cpp.o
[ 14%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/interprocess/signals.cpp.o
[ 14%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/interprocess/file_mapping.cpp.o
[ 15%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/interprocess/mmap_struct.cpp.o
[ 15%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/interprocess/file_mutex.cpp.o
[ 15%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/rpc/cli.cpp.o
[ 16%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/rpc/http_api.cpp.o
[ 16%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/rpc/json_connection.cpp.o
[ 16%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/rpc/state.cpp.o
[ 17%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/rpc/bstate.cpp.o
[ 17%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/rpc/websocket_api.cpp.o
[ 17%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/log/log_message.cpp.o
[ 18%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/log/logger.cpp.o
[ 18%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/log/appender.cpp.o
[ 18%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/log/console_appender.cpp.o
[ 19%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/log/file_appender.cpp.o
[ 19%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/log/gelf_appender.cpp.o
[ 19%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/log/logger_config.cpp.o
[ 19%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/crypto/_digest_common.cpp.o
[ 20%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/crypto/openssl.cpp.o
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/openssl.cpp: In constructor ‘fc::openssl_scope::openssl_scope()’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/openssl.cpp:33:33: warning: ‘void OPENSSL_config(const char*)’ is deprecated [-Wdeprecated-declarations]
           OPENSSL_config(nullptr);
                                 ^
In file included from /usr/include/openssl/evp.h:13:0,
                 from /home/zhr/bitshare/bitshares-core/libraries/fc/include/fc/crypto/openssl.hpp:4,
                 from /home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/openssl.cpp:1:
/usr/include/openssl/conf.h:92:1: note: declared here
 DEPRECATEDIN_1_1_0(void OPENSSL_config(const char *config_name))
 ^
[ 20%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/crypto/aes.cpp.o
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/aes.cpp: In destructor ‘fc::openssl_thread_config::~openssl_thread_config()’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/aes.cpp:432:32: warning: the address of ‘static long unsigned int fc::openssl_thread_config::get_thread_id()’ will never be NULL [-Waddress]
   if (CRYPTO_get_id_callback() == &get_thread_id)
                                ^
[ 20%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/crypto/crc.cpp.o
[ 21%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/crypto/city.cpp.o
[ 21%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/crypto/base36.cpp.o
[ 21%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/crypto/base58.cpp.o
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:69:24: error: invalid use of incomplete type ‘BIGNUM {aka struct bignum_st}’
 class CBigNum : public BIGNUM
                        ^~~~~~
In file included from /usr/include/openssl/bn.h:32:0,
                 from /home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:30:
/usr/include/openssl/ossl_typ.h:80:16: note: forward declaration of ‘BIGNUM {aka struct bignum_st}’
 typedef struct bignum_st BIGNUM;
                ^~~~~~~~~
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In constructor ‘CBigNum::CBigNum()’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:74:9: error: ‘BN_init’ was not declared in this scope
         BN_init(this);
         ^~~~~~~
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:74:9: note: suggested alternative: ‘BN_print’
         BN_init(this);
         ^~~~~~~
         BN_print
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In copy constructor ‘CBigNum::CBigNum(const CBigNum&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:79:9: error: ‘BN_init’ was not declared in this scope
         BN_init(this);
         ^~~~~~~
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:79:9: note: suggested alternative: ‘BN_print’
         BN_init(this);
         ^~~~~~~
         BN_print
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:80:30: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘BIGNUM* BN_copy(BIGNUM*, const BIGNUM*)’
         if (!BN_copy(this, &b))
                              ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:82:31: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘void BN_clear_free(BIGNUM*)’
             BN_clear_free(this);
                               ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘CBigNum& CBigNum::operator=(const CBigNum&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:89:30: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘BIGNUM* BN_copy(BIGNUM*, const BIGNUM*)’
         if (!BN_copy(this, &b))
                              ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In destructor ‘CBigNum::~CBigNum()’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:96:27: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘void BN_clear_free(BIGNUM*)’
         BN_clear_free(this);
                           ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In constructor ‘CBigNum::CBigNum(signed char)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:100:35: error: ‘BN_init’ was not declared in this scope
     CBigNum(signed char n)      { BN_init(this); if (n >= 0) setulong(n); else setint64(n); }
                                   ^~~~~~~
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:100:35: note: suggested alternative: ‘BN_print’
     CBigNum(signed char n)      { BN_init(this); if (n >= 0) setulong(n); else setint64(n); }
                                   ^~~~~~~
                                   BN_print
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In constructor ‘CBigNum::CBigNum(short int)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:101:35: error: ‘BN_init’ was not declared in this scope
     CBigNum(short n)            { BN_init(this); if (n >= 0) setulong(n); else setint64(n); }
                                   ^~~~~~~
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:101:35: note: suggested alternative: ‘BN_print’
     CBigNum(short n)            { BN_init(this); if (n >= 0) setulong(n); else setint64(n); }
                                   ^~~~~~~
                                   BN_print
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In constructor ‘CBigNum::CBigNum(int)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:102:35: error: ‘BN_init’ was not declared in this scope
     CBigNum(int n)              { BN_init(this); if (n >= 0) setulong(n); else setint64(n); }
                                   ^~~~~~~
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:102:35: note: suggested alternative: ‘BN_print’
     CBigNum(int n)              { BN_init(this); if (n >= 0) setulong(n); else setint64(n); }
                                   ^~~~~~~
                                   BN_print
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In constructor ‘CBigNum::CBigNum(int64_t)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:104:35: error: ‘BN_init’ was not declared in this scope
     CBigNum(int64_t n)          { BN_init(this); setint64(n); }
                                   ^~~~~~~
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:104:35: note: suggested alternative: ‘BN_print’
     CBigNum(int64_t n)          { BN_init(this); setint64(n); }
                                   ^~~~~~~
                                   BN_print
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In constructor ‘CBigNum::CBigNum(unsigned char)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:105:35: error: ‘BN_init’ was not declared in this scope
     CBigNum(unsigned char n)    { BN_init(this); setulong(n); }
                                   ^~~~~~~
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:105:35: note: suggested alternative: ‘BN_print’
     CBigNum(unsigned char n)    { BN_init(this); setulong(n); }
                                   ^~~~~~~
                                   BN_print
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In constructor ‘CBigNum::CBigNum(short unsigned int)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:106:35: error: ‘BN_init’ was not declared in this scope
     CBigNum(unsigned short n)   { BN_init(this); setulong(n); }
                                   ^~~~~~~
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:106:35: note: suggested alternative: ‘BN_print’
     CBigNum(unsigned short n)   { BN_init(this); setulong(n); }
                                   ^~~~~~~
                                   BN_print
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In constructor ‘CBigNum::CBigNum(unsigned int)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:107:35: error: ‘BN_init’ was not declared in this scope
     CBigNum(unsigned int n)     { BN_init(this); setulong(n); }
                                   ^~~~~~~
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:107:35: note: suggested alternative: ‘BN_print’
     CBigNum(unsigned int n)     { BN_init(this); setulong(n); }
                                   ^~~~~~~
                                   BN_print
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In constructor ‘CBigNum::CBigNum(uint64_t)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:109:37: error: ‘BN_init’ was not declared in this scope
     CBigNum(uint64_t n)           { BN_init(this); setuint64(n); }
                                     ^~~~~~~
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:109:37: note: suggested alternative: ‘BN_print’
     CBigNum(uint64_t n)           { BN_init(this); setuint64(n); }
                                     ^~~~~~~
                                     BN_print
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In constructor ‘CBigNum::CBigNum(const std::vector<unsigned char>&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:113:9: error: ‘BN_init’ was not declared in this scope
         BN_init(this);
         ^~~~~~~
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:113:9: note: suggested alternative: ‘BN_print’
         BN_init(this);
         ^~~~~~~
         BN_print
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘void CBigNum::setulong(long unsigned int)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:119:33: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_set_word(BIGNUM*, long unsigned int)’
         if (!BN_set_word(this, n))
                                 ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘long unsigned int CBigNum::getulong() const’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:125:32: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘long unsigned int BN_get_word(const BIGNUM*)’
         return BN_get_word(this);
                                ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘unsigned int CBigNum::getuint() const’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:130:32: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘long unsigned int BN_get_word(const BIGNUM*)’
         return BN_get_word(this);
                                ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘int CBigNum::getint() const’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:135:43: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘long unsigned int BN_get_word(const BIGNUM*)’
         unsigned long n = BN_get_word(this);
                                           ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:136:33: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_is_negative(const BIGNUM*)’
         if (!BN_is_negative(this))
                                 ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘void CBigNum::setint64(int64_t)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:174:37: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘3’ to ‘BIGNUM* BN_mpi2bn(const unsigned char*, int, BIGNUM*)’
         BN_mpi2bn(pch, p - pch, this);
                                     ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘void CBigNum::setuint64(uint64_t)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:201:37: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘3’ to ‘BIGNUM* BN_mpi2bn(const unsigned char*, int, BIGNUM*)’
         BN_mpi2bn(pch, p - pch, this);
                                     ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘void CBigNum::setvch(const std::vector<unsigned char>&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:217:46: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘3’ to ‘BIGNUM* BN_mpi2bn(const unsigned char*, int, BIGNUM*)’
         BN_mpi2bn(&vch2[0], vch2.size(), this);
                                              ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘std::vector<unsigned char> CBigNum::getvch() const’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:222:50: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_bn2mpi(const BIGNUM*, unsigned char*)’
         unsigned int nSize = BN_bn2mpi(this, NULL);
                                                  ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:226:32: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_bn2mpi(const BIGNUM*, unsigned char*)’
         BN_bn2mpi(this, &vch[0]);
                                ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘CBigNum& CBigNum::SetCompact(unsigned int)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:240:44: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘3’ to ‘BIGNUM* BN_mpi2bn(const unsigned char*, int, BIGNUM*)’
         BN_mpi2bn(&vch[0], vch.size(), this);
                                            ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘unsigned int CBigNum::GetCompact() const’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:246:50: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_bn2mpi(const BIGNUM*, unsigned char*)’
         unsigned int nSize = BN_bn2mpi(this, NULL);
                                                  ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:249:32: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_bn2mpi(const BIGNUM*, unsigned char*)’
         BN_bn2mpi(this, &vch[0]);
                                ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘std::__cxx11::string CBigNum::ToString(int) const’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:294:35: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘void BN_set_negative(BIGNUM*, int)’
         BN_set_negative(&bn, false);
                                   ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:297:29: error: cannot convert ‘CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_cmp(const BIGNUM*, const BIGNUM*)’
         if (BN_cmp(&bn, &bn0) == 0)
                             ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:299:32: error: cannot convert ‘CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_cmp(const BIGNUM*, const BIGNUM*)’
         while (BN_cmp(&bn, &bn0) > 0)
                                ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:301:54: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_div(BIGNUM*, BIGNUM*, const BIGNUM*, const BIGNUM*, BN_CTX*)’
             if (!BN_div(&dv, &rem, &bn, &bnBase, pctx))
                                                      ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:307:32: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_is_negative(const BIGNUM*)’
         if (BN_is_negative(this))
                                ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘bool CBigNum::operator!() const’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:322:31: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_is_zero(const BIGNUM*)’
         return BN_is_zero(this);
                               ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘CBigNum& CBigNum::operator+=(const CBigNum&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:327:35: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_add(BIGNUM*, const BIGNUM*, const BIGNUM*)’
         if (!BN_add(this, this, &b))
                                   ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘CBigNum& CBigNum::operator*=(const CBigNum&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:341:41: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_mul(BIGNUM*, const BIGNUM*, const BIGNUM*, BN_CTX*)’
         if (!BN_mul(this, this, &b, pctx))
                                         ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘CBigNum& CBigNum::operator<<=(unsigned int)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:360:41: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_lshift(BIGNUM*, const BIGNUM*, int)’
         if (!BN_lshift(this, this, shift))
                                         ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘CBigNum& CBigNum::operator>>=(unsigned int)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:371:28: error: cannot convert ‘CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_cmp(const BIGNUM*, const BIGNUM*)’
         if (BN_cmp(&a, this) > 0)
                            ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:377:41: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_rshift(BIGNUM*, const BIGNUM*, int)’
         if (!BN_rshift(this, this, shift))
                                         ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘CBigNum& CBigNum::operator++()’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:386:47: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_add(BIGNUM*, const BIGNUM*, const BIGNUM*)’
         if (!BN_add(this, this, BN_value_one()))
                                               ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In member function ‘CBigNum& CBigNum::operator--()’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:403:45: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_sub(BIGNUM*, const BIGNUM*, const BIGNUM*)’
         if (!BN_sub(&r, this, BN_value_one()))
                                             ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In function ‘const CBigNum operator+(const CBigNum&, const CBigNum&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:428:27: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_add(BIGNUM*, const BIGNUM*, const BIGNUM*)’
     if (!BN_add(&r, &a, &b))
                           ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In function ‘const CBigNum operator-(const CBigNum&, const CBigNum&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:436:27: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_sub(BIGNUM*, const BIGNUM*, const BIGNUM*)’
     if (!BN_sub(&r, &a, &b))
                           ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In function ‘const CBigNum operator-(const CBigNum&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:444:43: error: cannot convert ‘CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_is_negative(const BIGNUM*)’
     BN_set_negative(&r, !BN_is_negative(&r));
                                           ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In function ‘const CBigNum operator*(const CBigNum&, const CBigNum&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:452:33: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_mul(BIGNUM*, const BIGNUM*, const BIGNUM*, BN_CTX*)’
     if (!BN_mul(&r, &a, &b, pctx))
                                 ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In function ‘const CBigNum operator/(const CBigNum&, const CBigNum&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:461:39: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_div(BIGNUM*, BIGNUM*, const BIGNUM*, const BIGNUM*, BN_CTX*)’
     if (!BN_div(&r, NULL, &a, &b, pctx))
                                       ^
In file included from /home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:30:0:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In function ‘const CBigNum operator%(const CBigNum&, const CBigNum&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:470:10: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘2’ to ‘int BN_div(BIGNUM*, BIGNUM*, const BIGNUM*, const BIGNUM*, BN_CTX*)’
     if (!BN_mod(&r, &a, &b, pctx))
          ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In function ‘const CBigNum operator<<(const CBigNum&, unsigned int)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:478:33: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_lshift(BIGNUM*, const BIGNUM*, int)’
     if (!BN_lshift(&r, &a, shift))
                                 ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In function ‘bool operator==(const CBigNum&, const CBigNum&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:490:83: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_cmp(const BIGNUM*, const BIGNUM*)’
 perator==(const CBigNum& a, const CBigNum& b) { return (BN_cmp(&a, &b) == 0); }
                                                                      ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In function ‘bool operator!=(const CBigNum&, const CBigNum&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:491:83: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_cmp(const BIGNUM*, const BIGNUM*)’
 perator!=(const CBigNum& a, const CBigNum& b) { return (BN_cmp(&a, &b) != 0); }
                                                                      ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In function ‘bool operator<=(const CBigNum&, const CBigNum&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:492:83: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_cmp(const BIGNUM*, const BIGNUM*)’
 perator<=(const CBigNum& a, const CBigNum& b) { return (BN_cmp(&a, &b) <= 0); }
                                                                      ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In function ‘bool operator>=(const CBigNum&, const CBigNum&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:493:83: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_cmp(const BIGNUM*, const BIGNUM*)’
 perator>=(const CBigNum& a, const CBigNum& b) { return (BN_cmp(&a, &b) >= 0); }
                                                                      ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In function ‘bool operator<(const CBigNum&, const CBigNum&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:494:83: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_cmp(const BIGNUM*, const BIGNUM*)’
 operator<(const CBigNum& a, const CBigNum& b)  { return (BN_cmp(&a, &b) < 0); }
                                                                       ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In function ‘bool operator>(const CBigNum&, const CBigNum&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:495:83: error: cannot convert ‘const CBigNum*’ to ‘const BIGNUM* {aka const bignum_st*}’ for argument ‘1’ to ‘int BN_cmp(const BIGNUM*, const BIGNUM*)’
 operator>(const CBigNum& a, const CBigNum& b)  { return (BN_cmp(&a, &b) > 0); }
                                                                       ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In function ‘std::__cxx11::string EncodeBase58(const unsigned char*, const unsigned char*)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:525:48: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_div(BIGNUM*, BIGNUM*, const BIGNUM*, const BIGNUM*, BN_CTX*)’
         if (!BN_div(&dv, &rem, &bn, &bn58, pctx))
                                                ^
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp: In function ‘bool DecodeBase58(const char*, std::vector<unsigned char>&)’:
/home/zhr/bitshare/bitshares-core/libraries/fc/src/crypto/base58.cpp:575:42: error: cannot convert ‘CBigNum*’ to ‘BIGNUM* {aka bignum_st*}’ for argument ‘1’ to ‘int BN_mul(BIGNUM*, const BIGNUM*, const BIGNUM*, BN_CTX*)’
         if (!BN_mul(&bn, &bn, &bn58, pctx))
                                          ^
libraries/fc/CMakeFiles/fc.dir/build.make:1286: recipe for target 'libraries/fc/CMakeFiles/fc.dir/src/crypto/base58.cpp.o' failed
make[2]: *** [libraries/fc/CMakeFiles/fc.dir/src/crypto/base58.cpp.o] Error 1
CMakeFiles/Makefile2:177: recipe for target 'libraries/fc/CMakeFiles/fc.dir/all' failed
make[1]: *** [libraries/fc/CMakeFiles/fc.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by abitmore** [comment]
@ying2025 please follow these steps:
```
git clone https://github.com/bitshares/bitshares-core.git
cd bitshares-core
git checkout develop
git submodule update --init --recursive
# Then run `cmake` and `make`
```

**Comment by rsswlkr** [comment]
For some reason this node didn't trust the git ssl certificate/server, so I had to work around the error message. 

bitshares-core$ sudo git remote set-head origin --auto
fatal: unable to access 'https://github.com/bitshares/bitshares-core.git/': server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none
bitshares-core$ sudo export GIT_SSL_NO_VERIFY=1
sudo: export: command not found
**bitshares-core$ sudo git config --global http.sslverify false**
bitshares-core$ sudo git remote set-head origin --auto
origin/HEAD set to master

and carry on ...

**Comment by fvisconti** [comment]
I'm on Ubuntu 18.04
`openssl version`: `OpenSSL 1.1.0g  2 Nov 2017`
followed all the steps mentioned by @abitmore, but facing the issue as in #409, which is said to be fixed on Ubuntu 17.10. 
I configured with `cmake -DCMAKE_BUILD_TYPE=Release` as @jmjatlanta said.
Don't know exactly how to move (not even why this should work with Ubuntu 17.10 and not with 18.04).

**Comment by jmjatlanta** [comment]
On Ubuntu 18.04, things should work just fine if you are using the develop branch. In fact, it is slightly easier as there is no need to compile a specific boost version.

There must be some kind of problem (misconfiguration?) between doxygen and perl, if I am reading issue #409 correctly. I wasn't around when that issue appeared, so I can only speculate. Please post your build log and I'll see if I can help.

Note: Github messes with the formatting of the build log. I've found it better to post on something like pastebin.com and post a link here.

**Comment by fvisconti** [comment]
This is the last message in the build log.
In #409, they say to comment out lines in `CMakeLists.txt`, but I found they have been replaced  in the `develop` branch..

```bash
[ 81%] Generating api_documentation.cpp
Can't locate doxygen/perlmod/DoxyDocs.pm in @INC (you may need to install the doxygen::perlmod::DoxyDocs module) (@INC contains: /hom
e/francesco/WORK/XXX/bitshares-core/libraries/wallet /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.26.1 /usr/local/share/per
l/5.26.1 /usr/lib/x86_64-linux-gnu/perl5/5.26 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.26 /usr/share/perl/5.26 /usr/local/li
b/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at /home/francesco/WORK/XXX/bitshares-core/libraries/wallet/generate_api_docum
entation.pl line 6.
libraries/wallet/CMakeFiles/graphene_wallet.dir/build.make:62: recipe for target 'libraries/wallet/api_documentation.cpp' failed
make[2]: *** [libraries/wallet/api_documentation.cpp] Error 2
CMakeFiles/Makefile2:1826: recipe for target 'libraries/wallet/CMakeFiles/graphene_wallet.dir/all' failed
make[1]: *** [libraries/wallet/CMakeFiles/graphene_wallet.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by jmjatlanta** [comment]
If you don't want to create the documentation, you could probably comment out line 4 of ``libraries/wallet/CMakeLists.txt`` like so...

``# find_package( Doxygen )``

I haven't tested it, but that should skip the documentation creation process. You'll have to re-run cmake, then run make, and let me know the results.

**Comment by fvisconti** [comment]
Ok, going to.

**Comment by fvisconti** [comment]
Builded succesfully, many thanks.

**Comment by abitmore** [comment]
- [ ] if unable to find doxygen, build process should not fail, but skip the step gracefully.

**Comment by oxarbitrage** [comment]
closing by work done before and final updating the docs at https://github.com/bitshares/bitshares-core/pull/1212
some testing was done by the core team today with ubuntu 18.01. gcc 7.3.0, boost 1.65.1(found in default installations) and everything looks good.

**Comment by abitmore** [comment]
Still some work to be done before final mainnet release. OP updated.

**Comment by oxarbitrage** [comment]
ok, got it, thanks.

**Comment by abitmore** [comment]
Done.

**Comment by abitmore** [comment]
Does the doxygen issue still exist? Will this help? https://github.com/steemit/steem/pull/3119/commits/071f6a997357a116d96c0b0215473d3c20fd5316

**Comment by zhangbo-miuNEST** [comment]
> Does the doxygen issue still exist? Will this help? [steemit/steem@071f6a9](https://github.com/steemit/steem/commit/071f6a997357a116d96c0b0215473d3c20fd5316)

It works.
